### DISCUSSION TITLE
为什么 Linux 需要 Swapping - 面向信仰编程 · /whys-the-design-linux-swapping

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#632] [Announcements]
https://draveness.me/whys-the-design-linux-swapping/

**Comment by GavinCruise** [comment]
1. swap 提供的参数有 swap 使用的方式（文件或者分区），swap 的大小，是否开启以及行为。
2. 在可用性要求较高的场合，使用降级而不是停止服务的方式比较合适。
如有错漏，请不吝指正


**Comment by jeffrey4l** [comment]
现在 swap 有一个 zram 的方式，可以把内存进行一定程度的压缩，性能比放在 硬盘上面要好一些。

https://github.com/Nefelim4ag/systemd-swap

**Comment by 7768df1cf36e4ae79c6979d3ed4c38d5** [comment]
这个主题是什么呀

**Comment by draveness** [comment]
> 这个主题是什么呀

你的问题是什么..

**Comment by zqbake** [comment]
请问这里的swapping和你之前的文章 `为什么 Linux 需要虚拟内存` 中的页面交换是什么关系呢，是一个东西吗？

**Comment by draveness** [comment]
> 请问这里的swapping和你之前的文章 `为什么 Linux 需要虚拟内存` 中的页面交换是什么关系呢，是一个东西吗？

不是一个东西

**Comment by zqbake** [comment]
> > 请问这里的swapping和你之前的文章 `为什么 Linux 需要虚拟内存` 中的页面交换是什么关系呢，是一个东西吗？
> 
> 不是一个东西

那之间是什么关系呢？

**Reply by JOJO0527** [reply]
Difference between paging and swaping

**Comment by draveness** [comment]
> > > 请问这里的swapping和你之前的文章 `为什么 Linux 需要虚拟内存` 中的页面交换是什么关系呢，是一个东西吗？
> > 
> > 
> > 不是一个东西
> 
> 那之间是什么关系呢？

另一篇文章说的是页面调度

**Comment by zqbake** [comment]
> > > > 请问这里的swapping和你之前的文章 `为什么 Linux 需要虚拟内存` 中的页面交换是什么关系呢，是一个东西吗？
> > > 
> > > 
> > > 不是一个东西
> > 
> > 
> > 那之间是什么关系呢？
> 
> 另一篇文章说的是页面调度

你说 `为什么 Linux 需要虚拟内存` 这篇文章吗

**Comment by zstone12** [comment]
虚拟内存 和 Swapping 这两个概念存在什么联系呢？

**Comment by draveness** [comment]
> 虚拟内存 和 Swapping 这两个概念存在什么联系呢？

> Swapping uses virtual memory to copy contents in primary memory (RAM) to or from secondary memory (not directly addressable by the CPU, on disk). Swap space is often a dedicated disk partition that is used to extend the amount of available memory.

**Comment by yuchanns** [comment]
用kubeadm安装k8s的时候为什么会建议关闭swap？控制oom？

**Comment by draveness** [comment]
> @yuchanns 用kubeadm安装k8s的时候为什么会建议关闭swap？控制oom？

不希望触发 swap，该 OOM 的时候就 kill 掉，迁移到其他机器上
